kotsadm/web/src/components/apps/AppVersionHistory.jsx

```diff
@@ -314,28 +317,66 @@ class AppVersionHistory extends Component {
           return;
         }
       }
+      // prompt to make sure user wants to deploy
+      this.setState({
+        displayConfirmDeploymentModal: true,
+        versionToDeploy: version,
+        confirmIsRedeploy: false,
+        confirmIsRollback: false,
+      });
+      return;
+    } else { // force deploy is set to true so finalize the deployment
+      this.finalizeDeployment();
     }
-    await this.props.makeCurrentVersion(match.params.slug, version);
+  }
+
+  finalizeDeployment = async () => {
+    const { match, updateCallback } = this.props;
+    const { versionToDeploy } = this.state;
+    this.setState({ displayConfirmDeploymentModal: false });
+    await this.props.makeCurrentVersion(match.params.slug, versionToDeploy);
     await this.fetchKotsDownstreamHistory();
     this.setState({ versionToDeploy: null });
 
-    if (this.props.updateCallback) {
-      this.props.updateCallback();
+    if (updateCallback && typeof updateCallback === "function") {
+      updateCallback();
     }
   }
 
-  redeployVersion = async (version) => {
-    const { match, app } = this.props;
+  redeployVersion = (version, isRollback = false) => {
+    const { app } = this.props;
     const clusterSlug = app.downstreams?.length && app.downstreams[0].cluster?.slug;
     if (!clusterSlug) {
       return;
     }
 
-    await this.props.redeployVersion(match.params.slug, version);
+    // prompt to make sure user wants to redeploy
+    if (isRollback) {
+      this.setState({
+        displayConfirmDeploymentModal: true,
+        confirmIsRedeploy: false,
+        confirmIsRollback: true,
+        versionToDeploy: version,
+      });
+    } else {
+      this.setState({
+        displayConfirmDeploymentModal: true,
+        confirmIsRedeploy: true,
```

Comment:
how about also setting `confirmIsRollback` to `false` here? you could also use string type and merge those into 1 variable